### ISSUE TITLE
Guide: Julie Moronuki

### ISSUE THREAD

**Issue by argumatronic** [lambdaconf/winter-retreat-2017#3]
Hi, I'm Julie Moronuki, an author of [Haskell Programming from First Principles](http://haskellbook.com/). I have a background in linguistics and philosophy, including logic, and have been a teacher of all levels for nearly 20 years. My mission has always been to make difficult concepts, whether in German philosophy or X-bar syntax, comprehensible to everyone who wants to learn. I am excited to be a guide for the Winter Retreat this year.

I am relatively new to programming, but I came to love Haskell because of the type system and typeclasses and the pervasive monoids.  My developing interests are centered around natural language processing using typed lambda calculus. 

My first commitment is, and will remain, teaching. To that end, I am working on several projects to improve the documentation-and-examples situation around Haskell, as well as to make some of the underlying concepts of Haskell more accessible.

I have already had some inquiries about Haskell lessons for relative beginners at the Winter Retreat, something I'd be happy to do. Those lessons could be on foundational concepts ("how do we get from monoids to monads?") or "how do I make a (fill in the blank) in Haskell?" But I could also plan a workshop on computational semantics! 

In other words, I'm here to serve you. What do you want to learn and how can I help you get there?

**Please post your suggestions for me below! Vote on ideas by using the thumbs up / thumbs down emojis. Thanks in advance for all of your feedback, and I look forward to being a guide at LambdaConf Winter Retreat 2017!**

**Comment by mudphone** [comment]
I am a beginner Haskell programmer who has read through several books, and am about a third of the way through the Haskell Book. I am attempting to develop an intuition for working with typed languages, but it's slow going for me. Any Haskell sessions focused on teaching would be helpful for me. Also, if sufficiently geared toward beginners, there would be interest from at least one person learning to program.

**Comment by jdegoes** [comment]
My wife who is going through HaskellBook will also be in attendance, but doesn't have a Github account. She'd be interested in Haskell beginner-friendly content. 

Myself, I'd lean more toward NLP using lambda calculus. But I'm most definitely not representative of attendees so don't consider that. :)

**Comment by rezalesmana** [comment]
Haskell for beginners sound great. How about building small ecommerce web in Haskell?

**Comment by hirad** [comment]
I've worked my way through Learn You a Haskell, but I've stumbled in practice. Would love to learn a better way to practice my Haskell-fu to a point where I can actually use it. In other words, lessons for relative beginners sound great!

**Comment by purefunctions** [comment]
Suggested libraries for real-world backend use and related discussions.

**Comment by ErrorNullPointer** [comment]
NLP using lambda calculus would be awesome!

**Comment by astronuge** [comment]
I'd love to learn more foundational concepts of haskell

**Comment by ghost** [comment]
Would Lambda NLP be demonstrated in a particular language or strictly theory? very interested in it! but not sure what my engagement w Haskell will be over the retreat.

**Comment by gilligan** [comment]
I would basically enjoy any kind of workshop for developing some non-trivial project.

**Comment by argumatronic** [comment]
@soth-ing that's a good question. it would have depended on what people wanted, but based on people's responses here and via other media, i'm probably not going to do that as a workshop. so maybe we can talk about that during some of the non-workshop time. :)

**Comment by alexi21** [comment]
Hi there, I am a Haskell beginner and have been coding in Javascript using Ramda, RX, and other functional patterns for about six months. I have a PhD in philosophy (including German Idealism) and would be interested to hear about how you have used your background to inform your approach to programming. Cheers, Alex

**Comment by argumatronic** [comment]
# Applicative and Monadic Parsing (Two Hours)

The `Applicative` and `Monad` typeclasses allow for function application in the presence of some structure. They are similar in some respects, but they have important differences, differences that are not always well understood. In practice, many people end up using `Monad` in places where `Applicative` might be more suitable.

It's a difference that is perhaps best learned by doing, by using them in real code and comparing what can be done in applicative contexts that cannot be done in monadic contexts, and vice versa. One fun way to do this is through parsing.

We will begin this workshop with a very brief overview of the two typeclasses and a short introduction to parsing in general. We will use two Haskell parsing libraries, `optparse-applicative` and (probably) `trifecta`, as we explore the practical differences between applicative and monadic contexts. Finally, we will build a small application using `optparse-applicative`.  

The workshop will be aimed at people who can read some Haskell syntax and are familiar with the concepts of `Applicative` and `Monad`, but might not have much experience using them in practical applications. You do not need to already understand parsing to attend, and you do not need to have ever built a whole project in Haskell. By the end of this workshop, you will have! 

- What language? All code will be in Haskell.   
- Who is this for?  People who want to understand `Applicative` and `Monad`. People who want to learn more about parser combinators. People who have never built an entire application in Haskell before and want to know how to get started doing that. The second hour or so will be focused on a project that uses the parsing libraries we've been exploring and may be of interest even to people who have written programs in Haskell before but have never used `optparse-applicative`.
- Level: Somewhere on the border of [Fire Lubline](http://lambdaconf.us/downloads/documents/lambdaconf_slfp.pdf) and [Ice Skrig](http://lambdaconf.us/downloads/documents/lambdaconf_slfp.pdf).


@jdegoes

**Comment by argumatronic** [comment]
@alexi21 This is something I would *really* love to talk about, but I'm not sure how much more interest there would be in such a talk. I'd be happy to either talk with you about it individually or perhaps do an impromptu short talk if there are other people interested.
